sfge/src/main/java/com/salesforce/rules/fls/apex/operations/StandardConditionDecomposer.java

```diff
@@ -0,0 +1,133 @@
+                                    : seekingSatisfied));
+        } else if (seekingSatisfied) {
+            // All other vertex types are base cases, so we don't recurse. Instead, if this vertex
+            // is one that
+            // we're expecting to be satisfied, we should add it to our list.
+            results.add(vertex);
+        }
+        return results;
+    }
+
+    private static List<BaseSFVertex> decomposeBooleanExpression(
+            BooleanExpressionVertex vertex, boolean seekingSatisfied) {
+        List<BaseSFVertex> results = new ArrayList<>();
+        BaseSFVertex lhs = vertex.getLhs();
+        BaseSFVertex rhs = vertex.getRhs();
+        boolean eitherSideTrue = True.isLiterallyTrue(lhs) || True.isLiterallyTrue(rhs);
+        boolean eitherSideFalse = False.isLiterallyFalse(lhs) || False.isLiterallyFalse(rhs);
+        // If we're expecting an AND expression to be satisfied, then by extension we're also
+        // seeking for both sides to be satisfied.
+        boolean expectingSatisfiedAnd = vertex.isOperatorAnd() && seekingSatisfied;
+        // If we're expecting an OR expression to be satisfied and either side is equivalent to
+        // a literal false, then the other side must be satisfied.
+        boolean expectingSatisfiedOrFalse =
+                vertex.isOperatorOr() && seekingSatisfied && eitherSideFalse;
+        // If we're expecting an OR expression to be unsatisfied, then by extension we're
+        // also expecting each side to be unsatisfied.
+        boolean expectingUnsatisfiedOr = vertex.isOperatorOr() && !seekingSatisfied;
+        // If we're expecting an EQUALS expression to be satisfied and either side is true,
+        // the other side must be true.
+        boolean expectingSatisfiedEqualsTrue =
+                vertex.isOperatorEquals() && seekingSatisfied && eitherSideTrue;
+        // If we're expecting an EQUALS expression to be unsatisfied and either side is false,
+        // then the other side must be true.
+        boolean expectingUnsatisfiedEqualsFalse =
+                vertex.isOperatorEquals() && !seekingSatisfied && eitherSideFalse;
+        // If we're expecting a NOT-EQUALS expression to be satisfied and either side is false,
+        // then the other side must be true.
+        boolean expectingSatisfiedNotEqualsFalse =
+                vertex.isOperatorNotEquals() && seekingSatisfied && eitherSideFalse;
+        // If we're expecting a NOT-EQUALS expression to be unsatisfied and either side is true,
+        // then the other side must also be true.
+        boolean expectingUnsatisfiedNotEqualsTrue =
+                vertex.isOperatorNotEquals() && !seekingSatisfied && eitherSideTrue;
+
+        // TODO: This system is a little bit naive, and there are definitely some edge cases
+        //       that it won't support. In the fullness of time, we may wish to rework
+        //       this (and the code that uses it) into something a bit more intelligent.
```

Comment:
If you already know a few cases, could you add them to this comment for future reference?